tracing/src/main/java/com/palantir/tracing/CrossThreadSpan.java

```diff
@@ -0,0 +1,146 @@
+        Trace trace = Tracer.hasTraceId()
+                ? Tracer.getOrCreateCurrentTrace()
+                : Tracer.createTrace(Observability.UNDECIDED, Tracers.randomId());
+
+        Optional<String> parentSpanId = trace.top().map(OpenSpan::getSpanId);
+        OpenSpan openSpan1 = OpenSpan.of(
+                operation,
+                Tracers.randomId(),
+                SpanType.LOCAL,
+                parentSpanId,
+                trace.getOriginatingSpanId());
+
+        return new CrossThreadSpan(trace.getTraceId(), trace.isObservable(), new AtomicReference<>(openSpan1));
+    }
+
+    /**
+     * Complete the span. This method does not change any thread-local state, so successive operations on this
+     * thread may have different traceIds. If you want to finish the span and continue work related to
+     * this traceId use {@link #sibling}, or if you want to continue work related to this traceId without completing
+     * the span, use {@link #child}.
+     */
+    public void terminate() {
+        OpenSpan inProgress = consumeOpenSpan("CrossThreadSpan may only be terminated once");
+
+        // end the OpenSpan
+        Map<String, String> metadata = Collections.emptyMap();
+        Span span = Tracer.toSpan(inProgress, metadata, traceId);
+
+        Tracer.notifyObservers(span);
+    }
+
+    /**
+     * Finish the cross-thread span and start a new (thread local span) with the given operation name.
+     * NOTE: this is destructive, because any existing threadlocal tracing state is thrown away.
+     */
+    @MustBeClosed
+    public CloseableTracer sibling(String nextOperationName) {
+        OpenSpan inProgress = consumeOpenSpan("Can't call CrossThreadSpan#sibling if span has already been closed");
+
+        Map<String, String> metadata = Collections.emptyMap();
+        Span span = Tracer.toSpan(inProgress, metadata, traceId);
+        Tracer.notifyObservers(span);
+
+        // we throw away the existing thread local state and replace it with our view of the world
+        Tracer.clearCurrentTrace();
+        Tracer.setTrace(Trace.of(observable, traceId));
+
+        return CloseableTracer.startSpan(nextOperationName, inProgress.getParentSpanId(), SpanType.LOCAL);
+    }
+
+    /**
+     * Start a new span parented to this current in-progress CrossThreadSpan.
+     * NOTE: this is destructive, because any existing threadlocal tracing state is thrown away.
+     */
+    @MustBeClosed
+    public CloseableTracer child(String nextOperationName) {
+        Tracer.clearCurrentTrace();
+
+        OpenSpan inProgress = accessOpenSpan("Can't create child after terminate has been called");
+
+        Tracer.setTrace(Trace.of(observable, traceId));
+
+        return CloseableTracer.startSpan(nextOperationName, Optional.of(inProgress.getSpanId()), SpanType.LOCAL);
+    }
+
+    /**
+     * Unusual method - returns a new {@link CrossThreadSpan} parented to this instance. Only necessary if you want
```

Comment:
would be helpful to include an example in the docs of when to use this, it's not obvious how to select `child` or this method, both result in a parented CrossThreadSpan?